api-inference-community/docker_images/sklearn/app/pipelines/structured_data_classification.py

```diff
@@ -0,0 +1,37 @@
+from typing import Dict, List, Union
+
+import joblib
+import numpy as np
+import pandas as pd
+from app.pipelines import Pipeline
+from huggingface_hub import cached_download, hf_hub_url
+
+
+DEFAULT_FILENAME = "sklearn_model.joblib"
+
+
+class StructuredDataClassificationPipeline(Pipeline):
+    def __init__(self, model_id: str):
+        # TODO: Obtain expected column names from repo.
+        self.model = joblib.load(
```

Comment:
As you mentioned, this is arbitrary code execution.

Currently the pods do run in a privileged environment that have access to the main HF DB.
I am firmly against opening this pandora's box.

It is also hard to sandbox and make it safer. This is why we have `spaces`.

On a more usability concern, because it uses pickle, a lot of users are referring custom code in their scikit pipeline:
For instance: 
https://stackoverflow.com/questions/57888291/how-to-properly-pickle-sklearn-pipeline-when-using-custom-transformer

So this won't even make arbitrary scikit pipeline usable on the hub.

I think scikit was not designed with serialization and arbitrary deployment in mind which is perfectly fine, but also not a great fit for widgets on the hub. Couldn't we just **not** enable widgets there ?

Reply:
Answer about usability and arbitrary sklearn pipeline usage. 
* You're right that custom transforms does not work with joblib.
* But there seems to be alternatives such as `dill` and `cloudpickle`. I just tried [cloudpickle](https://github.com/cloudpipe/cloudpickle) to save a pipeline with custom transforms and later on successfully load and execute the pipeline.